packages/dcos-integration-test/extra/test_endpoints.py

```diff
@@ -76,10 +77,15 @@ def test_if_uiconfig_is_available(cluster):
 
 
 def test_if_dcos_history_service_is_up(cluster):
-    r = cluster.get('/dcos-history-service/ping')
 
-    assert r.status_code == 200
-    assert 'pong' == r.text
+    @retry(stop_max_delay=20000, wait_fixed=500)
```

Comment:
Practically if we want to this "right" then we need to hit every dcos-history-service directly. Doing that requires being on every master and doing a ping directly. Making it work across open and enterprise is complicated. That's not feasible for fixing here / now. 

Doing something like "check statistically" I don't think adds any value here. Our goal is just "make sure the history service is up and running". This should produce a log of the failures + those messages, the contents to help people debug.

The issue here is purely that sometimes the history buffer has never once pulled state when the worker gets hit and pulls the state out. We could make the checks a little more tuned down, but practically if there's a  major error it'll still show up, we're not waiting for a ton of time, and we definitely do not ever wait for "everything to actually wait for up". Making things wait for up would fix this and a bunch of other issues, but is a very out of scope project for now.